server/api/impl/store/manifest.go

```diff
@@ -0,0 +1,304 @@
+// Copyright (c) 2020-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package store
+
+import (
+	"crypto/sha1" // nolint:gosec
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/pkg/errors"
+
+	"github.com/mattermost/mattermost-plugin-apps/apps"
+	"github.com/mattermost/mattermost-plugin-apps/awsclient"
+	"github.com/mattermost/mattermost-plugin-apps/server/api"
+	"github.com/mattermost/mattermost-plugin-apps/server/utils"
+	"github.com/mattermost/mattermost-plugin-apps/server/utils/httputils"
+)
+
+type manifestStore struct {
+	*Store
+
+	mutex sync.RWMutex
+
+	global  map[apps.AppID]*apps.Manifest
+	local   map[apps.AppID]*apps.Manifest
+	builtin map[apps.AppID]*apps.Manifest
+}
+
+var _ api.ManifestStore = (*manifestStore)(nil)
+
+func (s *manifestStore) InitBuiltin(manifests ...*apps.Manifest) {
+	s.mutex.Lock()
+	if s.builtin == nil {
+		s.builtin = map[apps.AppID]*apps.Manifest{}
+	}
+	for _, m := range manifests {
+		s.builtin[m.AppID] = m
+	}
+	s.mutex.Unlock()
+}
+
+func (s *manifestStore) InitGlobal(awscli awsclient.Client) error {
```

Comment:
I don't like the coupling here. Why should a store have a knowledge of the aws?

Reply:
'Cause the list needs to be initialized. I can move the "InitGlobal" into a function in the `Admin` package, and then have an `InitGlobal` in the store take a map, but really why? This is a 1-time function, and it's passed a client to do its job. manifestStore does not hold on to AWS.